### PR TITLE
stm32wba: Add flash manager driver for RF coexistence

### PR DESCRIPTION
**Pull request by erwango**
To address flash coexistence issue, implement a dedicated flash driver using Flash manager API provided by STM32WBA Cube library

### CODE REVIEW

**Review by FRASTM — COMMENTED** [review]
LGTM

File: drivers/flash/flash_stm32wba_fm.c [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+}
+
+void FM_ProcessRequest(void)
+{
+	k_work_submit_to_queue(&ble_ctlr_work_q, &fm_work);
+}
+
+void FM_BackgroundProcess_Entry(struct k_work *work)
+{
+	FM_BackgroundProcess();
```
Comment (FRASTM) [inline drivers/flash/flash_stm32wba_fm.c:54]: don't you need `ARG_UNUSED(work);`

**Review by FRASTM — DISMISSED** [review]

File: drivers/flash/flash_stm32wba_fm.c (hunk 1 above)
Comment (erwango) [reply drivers/flash/flash_stm32wba_fm.c:57]: Right, added

**Review by FRASTM — APPROVED** [review]

**Comment by erwango** [comment]
@aescolar Any idea why BabbleSim Tests are failing? It doesn't seem related to my changes.

**Review by gautierg-st — APPROVED** [review]

**Review by Desvauxm-st — APPROVED** [review]

**Comment by aescolar** [comment]
> @aescolar Any idea why BabbleSim Tests are failing? It doesn't seem related to my changes.

That test seems to be failing on main
@PavelVPV @alxelax @Andrewpini  ^^

**Review by Laczen — APPROVED** [review]
This seems a very welcome addition.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**